src/Migrations/HSLMigration.php

```diff
@@ -0,0 +1,684 @@
+      if ($new_node === null) {
+        continue;
+      }
+
+      // we know we're rewriting the node, so now we know we need the namespace
+      $found_namespaces[] = $namespace;
+
+      // replace it in the ast
+      $root = $root->replace($node, $new_node);
+
+      // potentially change adjacent expressions to check for null instead of false
+      if ($replace_config['replace_false_with_null'] ?? false) {
+        $root = $this->maybeChangeFalseToNull($root, $new_node);
+      }
+    }
+
+    if (C\count($found_namespaces) === 0) {
+      return $root;
+    }
+
+    // add "use namespace" declarations at the top if they aren't already present
+    $declarations =
+      vec($root->getDescendantsOfType(INamespaceUseDeclaration::class));
+    list($hsl_declarations, $suffixes) =
+      $this->findUseDeclarations($declarations);
+
+    $count_before = C\count($suffixes);
+
+    // add new suffixes to the current list of suffixes
+    $suffixes = Keyset\union($suffixes, $found_namespaces);
+
+    // added any new suffixes?
+    if (C\count($suffixes) === $count_before) {
+      return $root;
+    }
+
+    // remove any current use statements for HH\Lib\* namespaces, we'll group them together
+    $root = $root->removeWhere(
+      ($node, $parents) ==> C\contains($hsl_declarations, $node),
+    );
+
+    // build a possibly grouped namespace use declaration
+    $new_namespace_use_declaration = $this->buildUseDeclaration($suffixes);
+
+    // insert the nmew node: skip the <?hh sigil and namespace declaration if present,
+    // then insert before the first declaration that remains
+    foreach ($root->getChildren()['declarations']->getChildren() as $child) {
+      if ($child instanceof MarkupSection) {
+        continue;
+      }
+
+      if ($child instanceof NamespaceDeclaration) {
+        $body = $child->getBody();
+        // namespace Foo; style declaration, skip over it
+        if ($body instanceof NamespaceEmptyBody) {
+          continue;
+        }
+        // namespace Foo { style declaration
+        // insert the use statement inside the braces, before the first child
+        invariant($body instanceof NamespaceBody, 'expected NamespaceBody');
+        $first_child = $body->getDeclarationsx()->getChildren() |> C\firstx($$);
+        return
+          $root->insertBefore($first_child, $new_namespace_use_declaration);
+      }
+      return $root->insertBefore($child, $new_namespace_use_declaration);
+    }
+
+    invariant(false, 'should not fail to insert new node');
+  }
+
+  protected function resolveIntegerArgument(EditableNode $node): ?int {
+    if ($node instanceof LiteralExpression) {
+      $expr = $node->getExpression();
+      if ($expr instanceof DecimalLiteralToken) {
+        return (int)$expr->getText();
+      }
+
+      // a literal 0 shows as octal
+      if ($expr instanceof OctalLiteralToken) {
+        return (int)$expr->getText();
+      }
+
+      return null;
+    } elseif (
```

Comment:
- space between `else` and `if`
- given the if block above returns, this could just be an `if`